### ISSUE TITLE
Merge current clang 3.6

### ISSUE THREAD

**Issue by jonthn** [clang-omp/clang#59]
Would it be possible to merge current clang branch 3.5 in order to get latest fixes ?

**Comment by alexey-bataev** [comment]
Hi, yes, of course. I'll try to update it ASAP

Best regards,

# Alexey Bataev

Software Engineer
Intel Compiler Team
Intel Corp.

20.01.2015 21:02, Jonathan пишет:

**Comment by howarthjw** [comment]
Wouldn't it make more sense to just jump straight to merging the clang 3.6 branch as 3.6.0 is due out on Feb 21st.

**Comment by alexey-bataev** [comment]
Yes, probably. Actually, we don't have much time for upgrade because our 
goal is to add full support for OpenMP to trunk. But I'll do my best to 
upgrade to the latest release version ASAP.

Best regards,

# Alexey Bataev

Software Engineer
Intel Compiler Team
Intel Corp.

07.02.2015 3:04, howarthjw пишет:

**Comment by tycho** [comment]
I agree with @howarthjw, I'd prefer to skip 3.5.1 and see a rebase on clang 3.6 once it's available.

**Comment by howarthjw** [comment]
3.6 is out now so we have a release to rebase on.

**Comment by alexey-bataev** [comment]
Thanks, I will do my best to rebase it ASAP, but can't give any promises 
when it will be completed.

Best regards,

# Alexey Bataev

Software Engineer
Intel Compiler Team
Intel Corp.

27.02.2015 23:05, howarthjw пишет:

**Comment by jonthn** [comment]
Hello Alexey,

Do you think a merge/update could happen within the next 2 weeks ?
Otherwise I'll have to use the current "old" 3.5 clang for my project

**Comment by alexey-bataev** [comment]
Hi, no, I'm not sure that I will update it to 3.6. :( Unfortunately, I'm 
a little bit busy with upstreaming of OpenMP to official clang trunk.
But you can try to use trunk-based repositories for clang/llvm if you 
need lates version of the compiler (repositories llvm_trunk, clang_trunk 
and compiler_rt-trunk)

Best regards,

# Alexey Bataev

Software Engineer
Intel Compiler Team
Intel Corp.

30.03.2015 10:10, Jonathan пишет:

**Comment by jonthn** [comment]
So does clang-omp/clang_trunk includes approx. the required changes to support OpenMP 3.1 + 4.0 ?
If that's the case I'll base my work on it :)

**Comment by alexey-bataev** [comment]
Yes, recently we've committed several bugfixes and I think it is stable now.

Best regards,

# Alexey Bataev

Software Engineer
Intel Compiler Team
Intel Corp.

30.03.2015 12:46, Jonathan пишет:

**Comment by jonthn** [comment]
Ok great :)
Thanks again, sorry for the trouble.

**Comment by alexey-bataev** [comment]
No problems

Best regards,

# Alexey Bataev

Software Engineer
Intel Compiler Team
Intel Corp.

30.03.2015 13:15, Jonathan пишет:

**Comment by howarthjw** [comment]
Alexey,
     I am confused about the contents of clang-trunk on
https://github.com/clang-omp/clang_trunk. Is that just another local mirror
of upstream clang trunk or do you maintain all of the clang-omp changes
from the clang-omp branch in that clang-trunk branch? If so, shouldn't I be
able to diff the git branch at https://github.com/clang-omp/clang_trunk and
https://github.com/llvm-mirror/clang for a commit just prior to the
branch-3.6 creation to be able to diff for the clang-omp changes required
for 3.6?
                Jack

**Comment by alexey-bataev** [comment]
Jack, this repo includes all changes from clang-omp repo, i.e. it is 
clang trunk + OpenMP from clang-omp. You can just take a diff a apply it 
on top of 3.6, you'll need to make some additional work to make it 
compatible with 3.6.
All trunk based repos are maintained mostly by Hal Finkel, I'm helping 
him with bug and compatibility fixes.

Best regards,

# Alexey Bataev

Software Engineer
Intel Compiler Team
Intel Corp.

30.03.2015 16:47, howarthjw пишет:

**Comment by howarthjw** [comment]
Alexey,
     Strangely, when I rolled the clang-omp clang-trunk tree and
llvm-mirrors clang-trunk back to the period when clang-3.6 was branched...

https://github.com/clang-omp/clang_trunk/commit/76e42e4ac33179b838de69d5ffdec55bfa6064d4

with

commit reset --hard 76e42e4ac33179b838de69d5ffdec55bfa6064d4

diffing the two trees produced an empty file (as if clang-omp clang-trunk
was simply a mirror of llvm-mirrors clang-trunk at that point). Have the
clang-omp changes only been migrated into clang-trunk very recently?
            Jack

**Comment by hfinkel** [comment]
Hi Jack,

You've done something wrong, but nevertheless, you can't roll back clang_trunk to where 3.6 was branched. Unfortunately, I stopped updating clang_trunk from upstream from around November of last year until only a few weeks ago. Then a lot of work was done to reanimate it and fix bugs that resulted. Only within the last several weeks is it truly usable again.

 -Hal

----- Original Message -----

> From: "howarthjw" notifications@github.com
> To: "clang-omp/clang" clang@noreply.github.com
> Sent: Tuesday, March 31, 2015 8:26:26 AM
> Subject: Re: [clang] Merge current clang 3.6 (#59)
> 
> Alexey,
> Strangely, when I rolled the clang-omp clang-trunk tree and
> llvm-mirrors clang-trunk back to the period when clang-3.6 was
> branched...
> 
> https://github.com/clang-omp/clang_trunk/commit/76e42e4ac33179b838de69d5ffdec55bfa6064d4
> 
> with
> 
> commit reset --hard 76e42e4ac33179b838de69d5ffdec55bfa6064d4
> 
> diffing the two trees produced an empty file (as if clang-omp
> clang-trunk
> was simply a mirror of llvm-mirrors clang-trunk at that point). Have
> the
> clang-omp changes only been migrated into clang-trunk very recently?
> Jack
> 
> On Mon, Mar 30, 2015 at 10:45 PM, Alexey Bataev
> notifications@github.com
> wrote:
> 
> > Jack, this repo includes all changes from clang-omp repo, i.e. it
> > is
> > clang trunk + OpenMP from clang-omp. You can just take a diff a
> > apply it
> > on top of 3.6, you'll need to make some additional work to make it
> > compatible with 3.6.
> > All trunk based repos are maintained mostly by Hal Finkel, I'm
> > helping
> > him with bug and compatibility fixes.
> > 
> > Best regards,
> > 
> > # Alexey Bataev
> > 
> > Software Engineer
> > Intel Compiler Team
> > Intel Corp.
> > 
> > 30.03.2015 16:47, howarthjw пишет:
> > 
> > > Alexey,
> > > I am confused about the contents of clang-trunk on
> > > https://github.com/clang-omp/clang_trunk. Is that just another
> > > local
> > > mirror
> > > of upstream clang trunk or do you maintain all of the clang-omp
> > > changes
> > > from the clang-omp branch in that clang-trunk branch? If so,
> > > shouldn't
> > > I be
> > > able to diff the git branch at
> > > https://github.com/clang-omp/clang_trunk and
> > > https://github.com/llvm-mirror/clang for a commit just prior to
> > > the
> > > branch-3.6 creation to be able to diff for the clang-omp changes
> > > required
> > > for 3.6?
> > > Jack
> > > 
> > > On Mon, Mar 30, 2015 at 3:15 AM, Alexey Bataev
> > > <notifications@github.com
> > > 
> > > wrote:
> > > 
> > > > Hi, no, I'm not sure that I will update it to 3.6. :(
> > > > Unfortunately,
> > > > I'm
> > > > a little bit busy with upstreaming of OpenMP to official clang
> > > > trunk.
> > > > But you can try to use trunk-based repositories for clang/llvm
> > > > if you
> > > > need lates version of the compiler (repositories llvm_trunk,
> > > > clang_trunk
> > > > and compiler_rt-trunk)
> > > > 
> > > > Best regards,
> > > > 
> > > > # Alexey Bataev
> > > > 
> > > > Software Engineer
> > > > Intel Compiler Team
> > > > Intel Corp.
> > > > 
> > > > 30.03.2015 10:10, Jonathan пишет:
> > > > 
> > > > > Hello Alexey,
> > > > > 
> > > > > Do you think a merge/update could happen within the next 2
> > > > > weeks ?
> > > > > Otherwise I'll have to use the current "old" 3.5 clang for my
> > > > > project
> > > > > 
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <https://github.com/clang-omp/clang/issues/59#issuecomment-87574912
> > > > > .
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > https://github.com/clang-omp/clang/issues/59#issuecomment-87576539.
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/clang-omp/clang/issues/59#issuecomment-87687124.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/clang-omp/clang/issues/59#issuecomment-87907844.
> 
> —
> Reply to this email directly or view it on GitHub .

## 

Hal Finkel
Assistant Computational Scientist
Leadership Computing Facility
Argonne National Laboratory

**Comment by howarthjw** [comment]
Hal,
     Now that clang-omp is branched as 3.5, any chance you might take a
stab at back porting the clang-omp changes from clang-trunk into a
clang-omp branch based on 3.6?
       Jack

**Comment by rgeary1** [comment]
+1 for a clang3.6 branch

**Comment by alexey-bataev** [comment]
Guys, I'm really sorry for delay with update to 3.6. I'll try to make 
it, but I can't give any promises about dates.

Best regards,

# Alexey Bataev

Software Engineer
Intel Compiler Team
Intel Corp.

11.04.2015 1:34, Richard Geary пишет:

**Comment by howarthjw** [comment]
Wouldn't it make more sense, at this point, to try aggregating all of the OPENMP commits from cfe-commits (made after 3.5's release) and apply these to stock clang 3.6 for a new clang-omp branch (rather than try to rebase the current clang-omp 3.5 branch on 3.6). It might be easier to identify and move the residual changes left between such a clang-omp 3.6 and existing clang-3.5 branch than trying to merge all of the accumulated changes already committed in cfe trunk since the 3.5 release.
